R/download_data.R

```diff
@@ -0,0 +1,106 @@
+#' Download the data files of specified type, zones, and dates
+#' 
+#' This function downloads the data files of the specified type, zones, dates and data version.
+#' @param type The type of data to download. Can be `"origin-destination"` (or ust `"od"`), or `"trips_per_person"` (or just `"tpp"`) for v1 data. For v2 data `"overnight_stays"` (or just `"os"`) is also available. More data types to be supported in the future. See respective codebooks for more information. **ADD CODEBOOKS! to the package**
+#' @param zones The zones for which to download the data. Can be `"districts"` (or `"dist"`, `"distr"`) or `"municipalities"` (or `"muni"`, `"municip"`) for v1 data. Additionaly, these can be `"large_urban_areas"` (or `"lau"`) for v2 data.
+#' @inheritParams spod_dates_argument_to_dates_seq
+#' @param data_dir The directory where the data is stored. Defaults to the value returned by `spod_get_data_dir()` which returns the value of the environment variable `SPANISH_OD_DATA_DIR` or a temporary directory if the variable is not set.
+#' @param quiet Logical. If `TRUE`, the function does not print messages to the console. Defaults to `FALSE`.
+#' @param return_output Logical. If `TRUE`, the function returns a character vector of the paths to the downloaded files. If `FALSE`, the function returns `NULL`.
+#' 
+#' @return A character vector of the paths to the downloaded files. Unless `return_output = FALSE`, in which case the function returns `NULL`.
+#' 
+#' @export 
+#' @examples
+#' \dontrun{
+#' # Download the origin-destination on district level for the a date range in March 2020
+#' spod_download_data(type = "od", zones = "districts", 
+#'                    date_range = c("2020-03-20", "2020-03-24"))
+#' 
+#' # Download the origin-destination on district level for select dates in 2020 and 2021
+#' spod_download_data(type = "od", zones = "dist", 
+#'                    dates_list = c("2020-03-20", "2020-03-24", "2021-03-20", "2021-03-24"))
+#' 
+#' # Download the origin-destination on municipality level using regex for a date range in March 2020 
+#' # (the regex will capture the dates 2020-03-20 to 2020-03-24)
+#' spod_download_data(type = "od", zones = "municip", 
+#'                    date_regex = "2020032[0-4]")
+#' }
+spod_download_data <- function(
```

Comment:
We should run an auto formatter, e.g. styler or styler.equals. That's for another day.

Reply:
at some point yes, btw, I am a proponent of using `<-` for assignment in R.

Reply:
No objections, I realise I'm in the minority here, most developers are also arrow users. #2